src/aspire/utils/resolution_estimation.py

```diff
@@ -0,0 +1,394 @@
+        # Bandwidth ranges from `[-1/pixel_size, 1/pixel_size]`, so the
+        # total bandwidth is `2*(1/pixel_size)`.
+
+        # Given a real space signal observed with `L` bins
+        # (pixels/voxels), each with a `pixel_size` in Angstrom, we can
+        # compute the width of a Fourier space bin to be the `Bandwidth
+        # / L = (2*(1/pixel_size)) / L`.  Thus the frequency at an index
+        # `k` is `freq_k = k * 2 * (1 / pixel_size) / L = k * 2 /
+        # (pixel_size * L)
+
+        # _freq(k) Units: 1 / (pixels * (Angstrom / pixel) = 1 / Angstrom
+        # Similar idea to wavenumbers (cm-1).  Larger is higher frequency.
+        return k * 2 / (self.L * self.pixel_size)
+
+    def plot(self, save_to_file=False):
+        """
+        Generates a Fourier correlation plot.
+
+        :param save_to_file: Optionally, save plot to file.
+            Defaults False, enabled by providing a string filename.
+            User is responsible for providing reasonable filename.
+            See `https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.savefig.html`.
+        """
+
+        # Construct x-axis labels
+        x_inds = np.arange(self.correlations.shape[-1])
+        freqs = self._freq(x_inds)
+        # TODO: handle zero freq better
+        with np.errstate(divide="ignore"):
+            freqs_angstrom = 1 / freqs
+
+        # Check we're asking for a reasonable plot.
+        stack = self.correlations.shape[: -self.dim]
+        if len(stack) > 2:
+            raise RuntimeError(
+                f"Unable to plot figure tables with more than 2 dim, stack shape {stack}. Try reducing to a simpler request."
+            )
+        if np.prod(stack) > 1:
+            raise RuntimeError(
+                f"Unable to plot figure tables with more than 1 figures, stack shape {stack}. Try reducing to a simpler request."
+            )
+
+        plt.figure(figsize=(8, 6))
+        plt.title(self._plot_title)
+        plt.xlabel("Resolution (Angstrom)")
+        plt.ylabel("Correlation")
+        plt.ylim([0, 1.1])
+        plt.plot(freqs_angstrom, self.correlations[0][0])
+        # Display cutoff
+        plt.axhline(
+            y=self.cutoff, color="r", linestyle="--", label=f"cutoff={self.cutoff}"
+        )
+        # Display resolution
+        plt.axvline(
+            x=self.estimated_resolution[0][0],
+            color="b",
+            linestyle=":",
+            label=f"Resolution={self.estimated_resolution[0][0]:.3f}",
+        )
+        # x-axis in decreasing
+        plt.gca().invert_xaxis()
+        plt.legend(title=f"Method: {self.method}")
+
+        if save_to_file:
+            plt.savefig(save_to_file)
+
+        plt.show()
+
+
+# The following are user facing classes, and simply wrap
+# `_FourierCorrelation` after assigning dimension `dim` and any
+# dimension specific variables.
+
+
+class FourierRingCorrelation(_FourierCorrelation):
+    """
+    See `_FourierCorrelation`.
+    """
```

Comment:
The pretty formulas you made in `_FourierCorrelation` don't appear in the docs. Maybe put that docstring here and say "See `FourierRing(Shell)Correlation`."  up in the `_FourierCorrelation` docstring.

Reply:
Right. Good catch. I totally overlooked that.  So I renamed to `FourierCorrelation` (rm'd `_`) and that populates in the docs now.  I intentionally did not add the base class to __init__ because its not something I would encourage importing and using directly.

There reason I don't want to put the formula all around is maintaining such things just constitutes make-work.  Those formula are already written in 3 places (two which I regret, once should be enough...).  Adding two additional would make 5.  Thats a lot of doc overhead.